ggml/src/ggml-vulkan/ggml-vulkan.cpp

```diff
@@ -8295,6 +8295,45 @@ template <> void init_pushconst_tensor_offsets(ggml_backend_vk_context * ctx, vk
     GGML_UNUSED(src2);
 }
 
+static vk_subbuffer ggml_vk_tensor_subbuffer(
+    const ggml_backend_vk_context * ctx, const ggml_tensor * tensor, bool support_incontiguous,
+    vk_buffer buffer = nullptr, size_t offset = 0) {
+
+    if (!buffer) {
+        auto buf_ctx = (ggml_backend_vk_buffer_context *)tensor->buffer->context;
+        buffer = buf_ctx->dev_buffer;
+        offset = vk_tensor_offset(tensor) + tensor->view_offs;
+    }
+    GGML_ASSERT(buffer != nullptr);
+
+    size_t misalign_bytes = offset & (ctx->device->properties.limits.minStorageBufferOffsetAlignment - 1);
+    offset &= ~misalign_bytes;
+
+    size_t size;
+    if (support_incontiguous) {
+        size = ggml_nbytes(tensor) + misalign_bytes;
+        if (offset + size >= buffer->size) {
```

Comment:
This replicates the branch here https://github.com/ggml-org/llama.cpp/blob/master/ggml/src/ggml-vulkan/ggml-vulkan.cpp#L8620-L8631

Why is this needed? The case `offset+size > buffer->size` sounds like it should never happen. In the case where they're equal, the branch computes the same value that's already there.

It might clamp the size to `maxStorageBufferRange` but that seems like it would likely produce wrong results and should rather assert/abort?

Reply:
The tests are passing with sufficient VRAM... sort of, I had to tone them down a bit since some cases ran out of system RAM (48GB) instead and I didn't want to switch to yet another machine. They still used >>4 GB buffers.